.circleci/config.yml

```diff
@@ -38,24 +38,28 @@ jobs:
       - checkout
       - config-path
       - restore_cache:
-          key: venv-{{ checksum "constraints.txt" }}-{{ checksum "requirements.txt" }}
+          key: venv-{{ checksum "dev-requirements.txt" }}-{{ checksum "requirements.txt" }}
       - create-venv
       - run:
-          name: Install codecov
+          name: Install requirements
           command: |
-            pip install -c constraints.txt codecov
+            pip install -r requirements.txt
       - run:
-          name: Install requirements
+          name: Install dev-requirements
+          command: |
+            pip install -r dev-requirements.txt
+      - run:
+          name: Install codecov
           command: |
-            pip install -c constraints.txt -r requirements.txt
+            pip install codecov==2.0.16
       - save_cache:
-          key: venv-{{ checksum "constraints.txt" }}-{{ checksum "requirements.txt" }}
+          key: venv-{{ checksum "dev-requirements.txt" }}-{{ checksum "requirements.txt" }}
           paths:
             - venv
       - run:
           name: Install package
           command: |
-            pip install -c constraints.txt -e .
+            pip install -e .
```

Comment:
Why do we not constraint?

Reply:
What do you want to constrain? Everything is installed already. 
I can constrain it, but it should not change anything.